src/m365/teams/commands/message/message-restore.spec.ts

```diff
@@ -0,0 +1,235 @@
+import assert from 'assert';
+import sinon from 'sinon';
+import auth from '../../../../Auth.js';
+import { cli } from '../../../../cli/cli.js';
+import { CommandInfo } from '../../../../cli/CommandInfo.js';
+import { Logger } from '../../../../cli/Logger.js';
+import { CommandError } from '../../../../Command.js';
+import request from '../../../../request.js';
+import { telemetry } from '../../../../telemetry.js';
+import { pid } from '../../../../utils/pid.js';
+import { session } from '../../../../utils/session.js';
+import { sinonUtil } from '../../../../utils/sinonUtil.js';
+import commands from '../../commands.js';
+import command from './message-restore.js';
+import { settingsNames } from '../../../../settingsNames.js';
+import { accessToken } from '../../../../utils/accessToken.js';
+import { teams } from '../../../../utils/teams.js';
+
+describe(commands.MESSAGE_RESTORE, () => {
+  const messageId = '1540911392778';
+  const teamId = '5f5d7b71-1161-44d8-bcc1-3da710eb4171';
+  const channelId = '19:00000000000000000000000000000000@thread.skype';
+  const teamName = 'Team Name';
+  const channelName = 'Channel Name';
+
+  let log: string[];
+  let logger: Logger;
+  let commandInfo: CommandInfo;
+
+  before(() => {
+    sinon.stub(auth, 'restoreAuth').resolves();
+    sinon.stub(telemetry, 'trackEvent').returns();
+    sinon.stub(pid, 'getProcessName').returns('');
+    sinon.stub(session, 'getId').returns('');
+    auth.connection.accessTokens[auth.defaultResource] = {
+      expiresOn: 'abc',
+      accessToken: 'abc'
+    };
```

Comment:
Let's replace this with `sinon.stub(accessToken, 'ensureDelegatedAccessToken').returns();

Reply:
guess you mean `assertDelegatedAccessToken`?

Reply:
Yes, sorry.